cni/pkg/ambient/informers.go

```diff
@@ -62,27 +62,20 @@ func (s *Server) EnqueueNamespace(o controllers.Object) {
 	matchAmbient := o.GetLabels()[constants.DataplaneMode] == constants.DataplaneModeAmbient
 	if matchAmbient {
 		log.Infof("Namespace %s is enabled in ambient mesh", namespace)
-		for _, pod := range s.pods.List(namespace, klabels.Everything()) {
+	} else {
+		log.Infof("Namespace %s is disabled from ambient mesh", namespace)
+	}
+	for _, pod := range s.pods.List(namespace, klabels.Everything()) {
```

Comment:
I think we may meet race here in case namespace deleting. For this case, the pod should be `Delete`

Reply:
Yeah we don't need to handle ns delete events